src/test/java/org/kairosdb/core/properties/PropertiesValidatorTest.java

```diff
@@ -0,0 +1,117 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kairosdb.core.properties;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.name.Names;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.kairosdb.core.exception.KairosDBException;
+import org.kairosdb.core.reporting.MetricReporterService;
+import org.kairosdb.datastore.cassandra.CassandraConfiguration;
+
+import java.util.Optional;
+
+/**
+ * Tests for {@link PropertiesValidator}.
+ *
+ * @author Max Wittek (https://github.com/wittekm)
+ */
+public final class PropertiesValidatorTest {
+    private static final int SOME_POSITIVE_INT = 1234;
+    @Rule
+    public ExpectedException exceptionRule = ExpectedException.none();
+
+    @Test
+    public void testWhenNothingSpecifiedValidatorPasses() throws KairosDBException {
+        ValidatorArgs args = new ValidatorArgs();
+
+        validatorWithInjection(args).validate();
+    }
+
+    @Test
+    public void testWhenOnlyAlignIsOnWithoutReporterTtlValidatorFails() throws KairosDBException {
+        exceptionRule.expect(KairosDBException.class);
+        exceptionRule.expectMessage("you must set kairosdb.reporter.ttl to a positive value");
+
+        ValidatorArgs args = new ValidatorArgs();
+        args.alignDatapointTtlWithTimestamp = Optional.of(true);
+
+        validatorWithInjection(args).validate();
+    }
+
+    @Test
+    public void testWhenForceTtlIsOnWithoutCassandraTtlValidatorFails() throws KairosDBException {
+        exceptionRule.expect(KairosDBException.class);
+        exceptionRule.expectMessage("you must set kairosdb.datastore.cassandra.datapoint_ttl to a positive value");
+
+        ValidatorArgs args = new ValidatorArgs();
+        args.forceDefaultDatapointTtl = Optional.of(true);
+
+        validatorWithInjection(args).validate();
+    }
+
+    @Test
+    public void testWhenAlignOffValidatorSucceeds() throws KairosDBException {
+        ValidatorArgs args = new ValidatorArgs();
+        args.reporterTtl = Optional.of(SOME_POSITIVE_INT);
+        args.cassandraDatapointTtl = Optional.of(SOME_POSITIVE_INT);
+
+        validatorWithInjection(args).validate();
+    }
+
+    @Test
+    public void testWhenAlignIsOnAndBothTtlsSpecifiedValidatorSucceeds() throws KairosDBException {
+        ValidatorArgs args = new ValidatorArgs();
+        args.alignDatapointTtlWithTimestamp = Optional.of(true);
+        args.reporterTtl = Optional.of(SOME_POSITIVE_INT);
+        args.cassandraDatapointTtl = Optional.of(SOME_POSITIVE_INT);
+
+        validatorWithInjection(args).validate();
+    }
+
```

Comment:
There should probably be a test for successful validation when forceDefaultDatapointTtl is on as well

Reply:
Done